pkg/token/server.go

```diff
@@ -96,6 +97,11 @@ func postRoleToken(ts *tokenService, w http.ResponseWriter, r *http.Request) {
 
 	// cache lookup (token TTL must >= 1 minute)
 	rToken := ts.roleTokenCache.Load(k)
+	// The case of not being subject to file output.
+	if rToken == nil {
+		k.WriteFileRequired = false
+		rToken = ts.roleTokenCache.Load(k)
+	}
```

Comment:
How about define a function inside the `TokenCache`
```go
// LoadToDoName loads a token from the given CacheKey.
// It first checks if the token is WriteFileRequired.
// If WriteFileRequired is true, it updates the cache with WriteFileRequired set to true.
// Otherwise, it simply returns the token found in the cache (which can be nil).
func (c *LockedTokenCache) LoadToDoName(k CacheKey) Token {
    // Prioritize WriteFileRequired == true:
    k.WriteFileRequired = true
    token := c.Load(k)
    if token != nil {
        return token
    }

    k.WriteFileRequired = false
    return c.Load(k)
}
```
and do this following instead:
```suggestion
	// The case of not being subject to file output.
	rToken := ts.roleTokenCache.LoadToDoName(k)
```

Reply:
fixed in https://github.com/AthenZ/k8s-athenz-sia/pull/131/commits/c2df53383b2bc00be4658a1416cae5b696b4697b